### PR TITLE
Implement authentication and authorization

### PR DESCRIPTION
**Pull request by seiyanuta**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,5 +1,31 @@
...
+
+  def current_team
+    if @current_team
+      return @current_team
+    end
+
+    team = User.find_by_name!(params[:team])
+    if team != current_user
+      raise "You're not allowed to access."
+    end
```
Comment (sh19910711) [inline app/controllers/application_controller.rb]: > ```diff
> +    team = User.find_by_name!(params[:team])
> +    if team != current_user
> +      raise "You're not allowed to access."
> +    end
> ```

It seems a bit difficult to catch what they are doing.

Comment (seiyanuta) [inline app/controllers/application_controller.rb]: Who's "they"?

Comment (sh19910711) [inline app/controllers/application_controller.rb]: The `they` are `User`, `team` and `current_user` :-)

We can add a method of the User class for the logic like `current_user.member?(params[:team])` or something, perhaps?

Comment (seiyanuta) [inline app/controllers/application_controller.rb]: This PR does not implement team things.

Comment (sh19910711) [inline app/controllers/application_controller.rb]: > This PR does not implement team things.

Got it :+1: Let's improve here when we create the team features later.

**Comment by sh19910711** [comment]
The other parts seem good to me.

**Pull request merged**